### PR TITLE
fix: add package parameter to generated asset provider method

### LINKED ISSUE #360: [BUG]: `package_parameter_enabled` is ignored for `AssetGenImage` "`provider`" method
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

5.2.0

### Command type

build_runner (Default)

### What happened?

When generating image assets this package generates an `AssetGenImage` class that is the base class for all images and that provides two methods:
* `image`, that provides a Flutter `Image.asset`
* `provider`, that provides an `ImageProvider`

Also the package accepts an additional output parameter named `package_parameter_enabled` that allow us to take profit for a packaged architecture assets generation as explained in [Flutter official assets doc](https://docs.flutter.dev/development/ui/assets-and-images#from-packages).

This works great for `image` method because its implementation use `keyName` getter as the asset name, which contains the full package path. 

However for some reason `provider` uses `_assetName` as its asset name, which does **not** contain the asset full path with the package. 

Also the `provider` implementation should probably propose a complete method with all the parameters from Flutter original `AssetImage` constructor.

The solution is really easy, I could propose a PR for it.

### Relevant a pubspec.yaml.

```yaml
name: myapp
description: A new Flutter package project.
version: 1.0.0
publish_to: none

environment:
  sdk: '>=2.19.0 <3.0.0'
  flutter: ">=3.7.0 <4.0.0"

dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^1.0.5
  flutter_svg: ^1.1.6

dev_dependencies:
  flutter_test:
    sdk: flutter
  build_runner: ^2.3.0
  flutter_gen_runner: ^5.1.0+1

flutter_gen:
  integrations:
    flutter_svg: true
  output: lib/assets/gen
  assets:
    enabled: true
    outputs:
      # Assets need to be referenced with the package option, see
      # https://docs.flutter.dev/development/ui/assets-and-images#from-packages
      package_parameter_enabled: true

flutter:
  uses-material-design: true
  assets:
    - assets/images/logo/
```

### Relevant log output

```shell
Example of generated `AssetGenImage`:


class AssetGenImage {
  const AssetGenImage(this._assetName);

  final String _assetName;

  Image image({
    Key? key,
    AssetBundle? bundle,
    ImageFrameBuilder? frameBuilder,
    ImageErrorWidgetBuilder? errorBuilder,
    String? semanticLabel,
    bool excludeFromSemantics = false,
    double? scale,
    double? width,
    double? height,
    Color? color,
    Animation<double>? opacity,
    BlendMode? colorBlendMode,
    BoxFit? fit,
    AlignmentGeometry alignment = Alignment.center,
    ImageRepeat repeat = ImageRepeat.noRepeat,
    Rect? centerSlice,
    bool matchTextDirection = false,
    bool gaplessPlayback = false,
    bool isAntiAlias = false,
    String? package = 'screens',
    FilterQuality filterQuality = FilterQuality.low,
    int? cacheWidth,
    int? cacheHeight,
  }) {
    return Image.asset(
      _assetName,
      key: key,
      bundle: bundle,
      frameBuilder: frameBuilder,
      errorBuilder: errorBuilder,
      semanticLabel: semanticLabel,
      excludeFromSemantics: excludeFromSemantics,
      scale: scale,
      width: width,
      height: height,
      color: color,
      opacity: opacity,
      colorBlendMode: colorBlendMode,
      fit: fit,
      alignment: alignment,
      repeat: repeat,
      centerSlice: centerSlice,
      matchTextDirection: matchTextDirection,
      gaplessPlayback: gaplessPlayback,
      isAntiAlias: isAntiAlias,
      package: package,
      filterQuality: filterQuality,
      cacheWidth: cacheWidth,
      cacheHeight: cacheHeight,
    );
  }

  ImageProvider provider() => AssetImage(_assetName);

  String get path => _assetName;

  String get keyName => 'packages/screens/$_assetName';
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by orevial**
Closes #360

## What does this change?

Make generated Image.provider method accepts an optional package name, which default to current package when using `package_parameter_enabled` flag

Fixes #360 🎯

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist:

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Make sure to open a GitHub issue as a bug/feature request before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
  - [x] Ensure the tests (`melos run unit:test`)
  - [x] Ensure the analyzer and formatter pass (`melos run format` to automatically apply formatting)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: packages/core/lib/generators/assets_generator.dart [hunk 1]
```diff
@@ -570,7 +570,16 @@ class AssetGenImage {
     );
   }
 
-  ImageProvider provider() => AssetImage(_assetName);
+  ImageProvider provider({
```
Comment (orevial) [inline packages/core/lib/generators/assets_generator.dart:573]: This is the one and only real change : 
* `provider` method now sets the package if the flag is enabled
* `provider` method now accepts arguments that mirror the `AssetImage` constructor

Note that the rest of the files are only updates of tests and examples

**Review by spilioio — APPROVED** [review]

**Comment by wasabeef** [comment]
@orevial
Thank you. I plan to release a new version today.

**Comment by blaugold** [comment]
Thanks @wasabeef!

**Comment by wasabeef** [comment]
@orevial @blaugold @spilioio 

I just released `v5.3.0`.  Sorry I kept you waiting.

**Pull request merged**